src/wweb/middleware/consumerCredits.ts

```diff
@@ -0,0 +1,71 @@
+import { Message } from 'whatsapp-web.js';
+
+import COMMANDS from '../onMessage/commands';
+import GroupModel from '../../group/groupModel';
+
+const getCreditsDiscountPerCommand = (command: string): number => {
+  if (command === COMMANDS.GPT.name) {
+    return 8;
+  }
+
+  if (command === COMMANDS.RESUME.name) {
+    return 10;
+  }
+
+  return 2;
+};
+
+export const consumerCredits = async (
+  msg: Message,
+): Promise<{ error: string | null }> => {
+  const groupId = msg.id.remote;
+  const command = msg.body;
+
+  const group = await GroupModel.findOne({
+    groupId,
+    removedAt: null,
+    isListening: true,
+  });
+
+  if (!group) {
+    return {
+      error: 'Group not found',
+    };
+  }
+
+  const creditsDiscount = getCreditsDiscountPerCommand(command);
+
+  if (group.credits < creditsDiscount) {
```

Comment:
some points here:
- how to handle concurrency, two users write at the same time
- how to debit the correct message and only one time
- this must be idempotent
- creditsDiscount? what does this mean? handle this like an account with balance and debits

Reply:
By start could have less validations